extension/new-tab.js

```diff
@@ -5,11 +5,18 @@
 // subject to an additional IP rights grant found at
 // http://polymer.github.io/PATENTS.txt
 
+
+// TODO(smalls) - there should be a better system of unique ids
+var faux_gid = 2000;
+
 /**
  * Load the current browsing data from all non-https tabs on all devices.
  */
 async function loadBrowsingData(arc, manifest) {
 
+  let entityKlass = manifest.findSchemaByName('Tab').entityClass();
+  let view = arc.createView(entityKlass.type.viewOf(), 'tabView'); // tabView could be 'anon'
```

Comment:
OK to start, but I think the views that contain the browsing data should be constructed in a manifest instead `... new Manifest(); manifest.newView(...)`. They would then get mapped/copied into the arc as necessary by the suggested recipe.

Reply:
Although that's probably 0.03 or 0.0.4 :P

Reply:
Oh good to know! I was going to ask about when to prefer one or the other - it's an interesting question, though, so I'll ask in the chat so we can all discuss.